src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/FundingModal/hooks.ts

```diff
@@ -0,0 +1,40 @@
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import useAsyncFunction from '~hooks/useAsyncFunction.ts';
+import { ActionTypes } from '~redux';
+import { type FundExpenditurePayload } from '~redux/sagas/expenditures/fundExpenditure.ts';
+import { type Expenditure } from '~types/graphql.ts';
+
+export const useGetFundingData = (expenditure: Expenditure) => {
+  const { colony } = useColonyContext();
+  const fundingItems = expenditure.slots.map((slot) => ({
+    key: slot.id,
+    amount: slot.payouts?.[0].amount || '0',
+    fee: slot.payouts?.[0].networkFee || '0',
+    tokenAddress: slot.payouts?.[0].tokenAddress || '',
+  }));
+
+  const fundExpenditure = useAsyncFunction({
+    submit: ActionTypes.EXPENDITURE_FUND,
+    error: ActionTypes.EXPENDITURE_FUND_ERROR,
+    success: ActionTypes.EXPENDITURE_FUND_SUCCESS,
+  });
```

Comment:
I think the preference would be to use `ActionButton` in `FundingModal`, this also allows you to handle loading states easier. This hook could then go away potentially.